packages/cashier/src/Components/CashierDefault/__tests__/cashier-default.spec.js

```diff
@@ -0,0 +1,234 @@
+        fireEvent.click(link);
+
+        expect(window.open).toHaveBeenCalledTimes(1);
+    });
+
+    it('should not show "Choose a way to fund your account" message if is_switching is false', () => {
+        const props = mockProps();
+        render(<CashierDefault {...props} is_switching />);
+
+        expect(screen.queryByText('Choose a way to fund your account')).not.toBeInTheDocument();
+    });
+
+    it('should not show "Choose a way to fund your account" message if accounts_list is an empty array', () => {
+        const props = mockProps();
+        render(<CashierDefault {...props} accounts_list={[]} />);
+
+        expect(screen.queryByText('Choose a way to fund your account')).not.toBeInTheDocument();
+    });
+
+    it('should not show "Choose a way to fund your account" message if is_landing_company_loaded is false', () => {
+        const props = mockProps();
+        render(<CashierDefault {...props} is_landing_company_loaded={false} />);
+
+        expect(screen.queryByText('Choose a way to fund your account')).not.toBeInTheDocument();
+    });
+});
+
+describe('<CashierDefaultDetails />', () => {
+    let props;
+    beforeEach(() => {
+        props = {
+            detail_click: jest.fn(),
+            detail_description: 'Deposit via the following payment methods:',
+            detail_header: 'Deposit via bank wire, credit card, and e-wallet',
+            is_dark_mode_on: false,
+            is_mobile: false,
+            detail_contents: [
+                {
+                    icons: [
+                        {
+                            dark: 'IcWalletCreditDebitDark',
+                            light: 'IcWalletCreditDebitLight',
+                        },
+                    ],
+                },
+            ],
+        };
+    });
+
+    it('should show the proper messages', () => {
+        render(<CashierDefaultDetails {...props} />);
+
+        expect(screen.getByText('Deposit via bank wire, credit card, and e-wallet')).toBeInTheDocument();
+        expect(screen.getByText('Deposit via the following payment methods:')).toBeInTheDocument();
+    });
+
+    it('should show contain the correct className, when detail_contents has icons', () => {
+        const { container } = render(<CashierDefaultDetails {...props} />);
+
+        expect(container.querySelector('.cashier-default-detail__array')).not.toBe(null);
+    });
+
+    it('should trigger onClick callback, when the user clicks on the block with details', () => {
+        const { container } = render(<CashierDefaultDetails {...props} />);
+
+        const details_block = container.querySelector('.cashier-default-detail__div');
+        fireEvent.click(details_block);
+
+        expect(props.detail_click).toHaveBeenCalledTimes(1);
+    });
+});
+
+describe('<CashierDefaultSideNote />', () => {
+    const textContentMatcher = text => {
+        return (content, node) => {
+            const hasText = node => node.textContent === text;
```

Comment:
@george-usynin-deriv Isn't it better to have another name for this second `node`?